### PR TITLE
Fixed up mysql and added init option to initialize with either pg or mysql

### PR DESCRIPTION
**Pull request by elorest**
fixed mysql migrations so they work now.
wrote mapping table for column types between databases.
added `--db mysql or pg` option to `init app`.
fixed docker setup to work with mysql as well.
added database options to readme and added some instructions on how to use.
modified specs to create test project inside of project folder and delete if when done.
added spec to test that database options worked.

### CODE REVIEW

File: src/templates/app/config/database.yml.ecr [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<%= @database %>:
+<% if @database == "mysql" -%>
```
Comment (drujensen) [inline src/templates/app/config/database.yml.ecr:2]: Is this conditional needed?  Your setting the environment variable DATABASE_URL in the docker-compose.yml.

**Review by drujensen — APPROVED** [review]

File: src/templates/app/docker-compose.yml.ecr [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    image: postgres
+    environment:
+      POSTGRES_PASSWORD: 'password'
+    volumes:
+      - 'pg:/var/lib/postgres/data'
+
+volumes:
+  pg:
+<% elsif @database == "mysql" %>
+services:
```
Comment (drujensen) [inline src/templates/app/docker-compose.yml.ecr:40]: I'm wondering if we can get rid of this duplicate code.  Only the db's and url's change.

File: src/templates/scaffold.cr [hunk 3]
```diff
@@ -9,18 +9,34 @@ module Kemalyst::Generator
...
+      @primary_key = primary_key
     end
 
     def database
       yaml_file = File.read("config/database.yml")
       yaml = YAML.parse(yaml_file)
       yaml.first.to_s
     end
+
+    def primary_key
```
Comment (drujensen) [inline src/templates/scaffold.cr:31]: nice!  this is way better.

File: src/templates/app/config/database.yml.ecr (hunk 1 above)
Comment (elorest) [reply src/templates/app/config/database.yml.ecr:2]: Technically I think that it should have defaults for both pg and mysql and eventually sqlite3 in the database.yml. Environment VAR DATABASE_URL should be loaded if it exists regardless of if something else is set in the database.yml.

File: src/templates/app/docker-compose.yml.ecr (hunk 2 above)
Comment (elorest) [reply src/templates/app/docker-compose.yml.ecr:40]: Yes that could be cleaned up. I'd like to change pg to db on both of them.

File: src/templates/app/config/database.yml.ecr (hunk 1 above)
Comment (elorest) [inline src/templates/app/config/database.yml.ecr]: https://github.com/drujensen/kemalyst-model/pull/25

Comment (elorest) [reply src/templates/app/config/database.yml.ecr:2]: https://github.com/drujensen/kemalyst-model/pull/25

File: src/templates/scaffold/db/migrations/{{timestamp}}_create_{{name}}.sql.ecr [hunk 4]
```diff
@@ -1,34 +1,8 @@
...
-<% when "pg" -%>
-CREATE TABLE <%= @name %>s (
-  id BIGSERIAL PRIMARY KEY,
-<% @fields.each do |field| -%>
-  <%= field.name %> <%= field.db_type %>,
-<% end -%>
-  created_at TIMESTAMP,
-  updated_at TIMESTAMP
-);
-<% else -%>
```
Comment (drujensen) [inline src/templates/scaffold/db/migrations/{{timestamp}}_create_{{name}}.sql.ecr:22]: Should we remove sqlite?

File: src/templates/app/config/database.yml.ecr (hunk 1 above)
Comment (drujensen) [reply src/templates/app/config/database.yml.ecr:2]: ok, not the biggest fan of DATABASE_URL magic in rails but understand.

**Comment by elorest** [comment]
I'll work on sqlite3 this weekend. I'll look at the docker-compose file a bit more when I have a chance.

**Pull request merged**